gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -128,61 +128,8 @@
 
   private final long gRPCReadMessageTimeout;
 
-  /**
-   * Used to open given file using item info
-   *
-   * @param stubProvider gRPC stub for accessing the Storage gRPC API
-   * @param storage store and retrieve data object
-   * @param errorExtractor ApiErrorExtractor instance to convert downstream error into appropriate
-   *     fs exception
-   * @param resourceId Identifier for the file to be opened
-   * @param watchdog Watchdog instance to monitor open streams
-   * @param readOptions readOptions fine-grained options specifying things like retry settings,
-   *     buffering, etc.
-   * @return gRPC read channel
-   * @throws IOException IO Error
-   */
-  public static GoogleCloudStorageGrpcReadChannel open(
-      StorageStubProvider stubProvider,
-      Storage storage,
-      ApiErrorExtractor errorExtractor,
-      StorageResourceId resourceId,
-      Watchdog watchdog,
-      GoogleCloudStorageReadOptions readOptions)
-      throws IOException {
-    return open(
-        stubProvider,
-        storage,
-        errorExtractor,
-        resourceId,
-        watchdog,
-        readOptions,
-        BackOffFactory.DEFAULT);
-  }
-
-  /**
-   * Used to open given file using item info
-   *
-   * @param stubProvider gRPC stub for accessing the Storage gRPC API
-   * @param storage store and retrieve data object
-   * @param itemInfo contains metadata information about the file
-   * @param readOptions readOptions fine-grained options specifying things like retry settings,
-   *     buffering, etc.
-   * @return gRPC read channel
-   * @throws IOException IOException on IO Error
-   */
-  static GoogleCloudStorageGrpcReadChannel open(
-      StorageStubProvider stubProvider,
-      Storage storage,
-      GoogleCloudStorageItemInfo itemInfo,
-      Watchdog watchdog,
-      GoogleCloudStorageReadOptions readOptions)
-      throws IOException {
-    return open(stubProvider, storage, itemInfo, watchdog, readOptions, BackOffFactory.DEFAULT);
-  }
-
   @VisibleForTesting
-  static GoogleCloudStorageGrpcReadChannel open(
+  GoogleCloudStorageGrpcReadChannel(
```

Comment:
There are multiple constructors with very many parameters - I think it could be better to introduce builder for this class.

Reply:
Builder with mandatory fields might be cumbersome here. Refactored, the constructer to 5 fields, (which will reduce to 4 once we migrate to grpc metadata apis) which I think is reasonable